[filbertphang] iP

filbertphang opened the pull request:
# duke
>  every 60 seconds in africa, a minute passes 🤔 

**_duke_** is a:
- glorified todo list
- that i made
- for cs2103t

## commands
here is a list of the available commands:
1. `todo <description>`
2. `deadline <description> /by <deadline>`
3. `event <description> /from <start date> /to <end date>`
4. `list`
5. `mark <index>`
6. `unmark <index>`
7. `delete <index>`
8. `bye`
9. `archive` OR `archive /name <archive file name>`

## todo
- [x] update this description with [GFMD](https://github.github.com/gfm/)
- [ ] do the first PR review
- [ ] do the second PR review
- [ ] add increments: ` A-CheckStyle`, `Level-10`, `A-Varargs`

## usage
after downloading the files, just run
```console
filbert@LAPTOP-D789KE:~$  java -jar Duke.jar
```

Commit:
Add Gradle support

filbertphang committed:
Update .gitignore to ignore class files

filbertphang committed:
Implement Level 1

filbertphang committed:
Add prettyPrint function

filbertphang committed:
Add .prettierrc for code formatting

filbertphang committed:
Implement Level 2

filbertphang committed:
Add Task class

filbertphang committed:
Update Task with mark/unmark methods

filbertphang committed:
Implement Level 3

filbertphang committed:
Add subclasses ToDo, Deadline, and Event

filbertphang committed:
Add argument parsing; Implement adding ToDo

filbertphang committed:
Add override annotation in subclasses of Task

filbertphang committed:
Implement Level 4

filbertphang committed:
Add one sample testcase for Automated Text UI Testing

filbertphang committed:
Add Duke exceptions

filbertphang committed:
Implement Level 5

filbertphang committed:
Implement Level 6

filbertphang committed:
Introduce Ui class

filbertphang committed:
Move Mark, Unmark, and List commands to use TaskList

filbertphang committed:
Move remaining commands to use TaskList

filbertphang committed:
Add command subclasses

filbertphang committed:
Update vscode workspace settings for Java formatter

filbertphang committed:
Implement Parser

filbertphang committed:
Add packages

filbertphang committed:
Remove printing ">" when bye command is given

filbertphang committed:
Add isDone constructor for Task and subclasses

filbertphang committed:
Implement serialization of tasks

filbertphang committed:
Edit gitignore and vscode formatter settings

filbertphang committed:
Rename isTerminating to doPromptUserInput

filbertphang committed:
Implement Level 7

filbertphang committed:
Implement Level 8

filbertphang committed:
Merge branch 'branch-Level-8'

filbertphang committed:
Merge branch 'add-gradle-support'

filbertphang committed:
Update Gradle config

filbertphang committed:
Improve exception handling for invalid save file

filbertphang committed:
Override equals() for command

filbertphang committed:
Override equals() for tasks

filbertphang committed:
Add JUnit tests

filbertphang committed:
Use relative paths for storage

filbertphang committed:
Add javadoc for package duke

filbertphang committed:
Add javadoc for package duke.task

filbertphang committed:
Add javadoc for package duke.exception

filbertphang committed:
Add javadoc for package duke.command

filbertphang committed:
Fix switch statements to use the proper identation

filbertphang committed:
Implement Level 9

filbertphang committed:
Merge branch 'branch-A-JavaDoc'

filbertphang committed:
Merge branch 'branch-A-CodingStandard'

filbertphang committed:
Merge branch 'branch-Level-9'

filbertphang committed:
Add javadoc for FindCommand

filbertphang committed:
Re-implement delete command

edures888 commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,150 @@
+     *                                  with numArgs given.
+     */
+    public static String[] getArgs(String commandName, String argString, int numArgs)
+
```
Small detail, this could be an extra newline?

filbertphang replied:
good catch, thanks

edures888 commented on src/main/java/duke/Ui.java:
```diff
@@ -0,0 +1,91 @@
+public class Ui {
+
+    private Scanner scanner;
+    private StringBuilder msg;
```
Another small detail, maybe it could be better to use the full word for the variable?

filbertphang replied:
sure, will edit

edures888 commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,150 @@
+        int index = -1;
+        String[] args = null;
+
+        // create commands depending on the command name
```
I like how the logical units are clearly separated!

edures888 commented on src/main/java/duke/Ui.java:
```diff
@@ -0,0 +1,91 @@
+
+    private Scanner scanner;
+    private StringBuilder msg;
+    private boolean doPromptUserInput;
```
Not sure if this helps, maybe naming it as `toPromptUserInput` could be clearer?

filbertphang replied:
actually on second thought maybe it should be `shouldPromptUserInput` 🤔

edures888 reviewed (commented):
I think the code follows most of the Java coding standards!

filbertphang committed:
rename "msg" to "message"

filbertphang committed:
removed redundant newline

filbertphang committed:
add checkstyle to project

filbertphang committed:
configure vscode to remove trailing whitespace

filbertphang committed:
remove trailing whitespace from files

filbertphang committed:
refactor code to adhere to checkstyle

filbertphang committed:
Merge branch 'branch-A-CheckStyle'

filbertphang committed:
include javafx in gradle build

filbertphang committed:
initialize javafx gui components

filbertphang committed:
configure sizing and positioning when initializing stage

filbertphang committed:
bind send button handler to call duke.run()

filbertphang committed:
add images for user and duke

filbertphang committed:
add dialogbox component

filbertphang committed:
update handler to echo user text using a dialogbox

filbertphang committed:
add display for messages by both parties

filbertphang committed:
resize dialogbox labels

filbertphang committed:
re-implement program exit with "bye" command

filbertphang committed:
refactor to remove redundant methods; add javadoc

filbertphang committed:
add javadoc

filbertphang committed:
Merge branch 'branch-Level-10'

filbertphang committed:
remove text ui test

dawg420 commented on src/main/java/duke/DialogBox.java:
```diff
@@ -0,0 +1,67 @@
+    private Label text;
+    private ImageView displayPicture;
+
+    /**
+     * Creates a DialogBox with a given text string and image.
+     *
+     * @param text  The message to be displayed
+     * @param image The image to be displayed
+     */
```
I like how your javadoc comments are concise and to the point.

dawg420 commented on src/main/java/duke/Duke.java:
```diff
@@ -19,44 +23,36 @@ public class Duke {
-        String rawInput;
+    public void run(String rawInput) {
         Command command;
         boolean isExit = false;
```
Perhaps

dawg420 commented on src/main/java/duke/DialogBox.java:
```diff
@@ -0,0 +1,67 @@
+public class DialogBox extends HBox {
+
+    // user images
+    private static final Image userImage = new Image(DialogBox.class.getResourceAsStream("/images/red.png"));
+    private static final Image dukeImage = new Image(DialogBox.class.getResourceAsStream("/images/cyan.png"));
```
Constants should be capitalised

filbertphang replied:
good catch, thanks

dawg420 reviewed (commented):
Generally every section of the code is very documented, easy to understand, and concise!

filbertphang committed:
rename user image and duke image to respect coding standards

filbertphang committed:
remove redundant imports in Ui

filbertphang committed:
add assertions

several assertions have been added to sanity-check the code.
most addditions are in TaskList, which check whether:
 - given indices for mark/unmark/delete tasks are valid
 - add/delete task operations work as intended (by checking tasklist
 size)

other changes include:
 - checking whether clearMessage() in Ui properly clears messages

filbertphang committed:
improve code quality

general code quality across the board has been improved, do see below
for the breakdown of individual files

Parser.java
 - extract flags for todo, deadline, and event commands into constants
 in their respective classes
 - extracted boolean condition in getArgs to its own variable to improve
 readability

DeadlineCommand.java, EventCommand.java, ToDoCommand.java
 - extract flags into respective classes (see changes for Parser.java)

Task.java
 - extract task status icons into constants

filbertphang committed:
Merge pull request #2 from filbertphang/branch-A-CodeQuality

improve code quality

filbertphang committed:
Merge branch 'master' into branch-A-Assertions

filbertphang committed:
Merge pull request #1 from filbertphang/branch-A-Assertions

add assertions

filbertphang committed:
implement archive command

filbertphang committed:
parse archive command

filbertphang committed:
update gitignore to remove archived files

filbertphang committed:
fix crlf

filbertphang committed:
Merge branch 'branch-C-Archive'

filbertphang committed:
add javadoc for archive command

filbertphang committed:
add todo, deadline, and event to user guide

filbertphang committed:
add 'mark', 'unmark', and 'delete' to user guide

filbertphang committed:
add 'list', 'find', 'archive', and 'bye' to user guide

filbertphang committed:
add quick start to user guide

filbertphang committed:
improve code quality

filbertphang committed:
replace user and duke avatars

filbertphang committed:
add UI screenshot

filbertphang committed:
Merge branch 'branch-A-UserGuide'

filbertphang committed:
add whitespace before screenshot

filbertphang committed:
Merge branch 'branch-A-UserGuide'

filbertphang committed:
fix gradle config to correctly build jar
